snudown.h

```diff
@@ -0,0 +1,118 @@
+/* snudown.h - header for snudown */
```

Comment:
Can you guys explain maintenance problems caused by including definition in .h?  

Is it because it would be difficult to use different implementation of the functions in case its needed (Such as slightly different implementation in fuzzer code, or somewhere else?)